digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java

```diff
@@ -202,4 +202,25 @@ public ArrayValue next()
             throw Throwables.propagate(ex.getCause());
         }
     }
+
+    public TDJobSummary checkStatus()
+    {
+        try {
+            return ensureRunningOrSucceeded();
+        }
+        catch (TDJobException ex) {
+            try {
+                TDJob job = getJobInfo();
+                String message = job.getCmdOut() + "\n" + job.getStdErr();
+                throw new TaskExecutionException(message, buildExceptionErrorConfig(ex));
+            }
+            catch (Exception getJobInfoFailed) {
+                getJobInfoFailed.addSuppressed(ex);
+                throw Throwables.propagate(getJobInfoFailed);
+            }
+        }
+        catch (InterruptedException ex) {
+            throw Throwables.propagate(ex);
```

Comment:
Yeah, unfortunately there's a lot of similar code in digdag where we're not properly restoring interrupted status. We should fix that everywhere.